pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java

```diff
@@ -267,6 +270,35 @@ private static void validateDistinctQuery(PinotQuery pinotQuery)
     }
   }
 
+  /*
+   * Throws an exception if the filter's rhs has NULL because:
+   * - Predicate evaluator and pruning do not have NULL support.
+   * - It is not meaningful to have NULL in the filter's rhs.
+   *   - For most of the filters (e.g. GREATER_THAN, LIKE), the rhs being NULL leads to no record matched.
+   *   - For IN, adding NULL to the rhs list does not change the matched records.
+   *   - For NOT IN, adding NULL to the rhs list leads to no record matched.
+   *   - For "= NULL" and "!= NULL", we have the IS_NULL and IS_NOT_NULL as workarounds.
```

Comment:
Semantically `= NULL` and `!= NULL` is not the same as `IS_NULL` and `IS_NOT_NULL`. They will always return `null` which is not really useful

Reply:
You are right. Corrected the function comment.